test/test_gpuarray.py

```diff
@@ -468,6 +468,49 @@ def test_arange(self):
         a = gpuarray.arange(12, dtype=np.float32)
         assert (np.arange(12, dtype=np.float32) == a.get()).all()
 
+    @mark_cuda_test
+    def test_stack(self):
+
+        orders = ["F", "C"]
+        input_dims = 2
+
+        for order in orders:
+            shape = (2, 2, 2)[:input_dims]
+            axis = -1 if order == "F" else 0
+
+            from numpy.random import default_rng
+            rng = default_rng()
+            x_in = rng.random(size=shape)
+            y_in = rng.random(size=shape)
+            x_in = x_in if order == "C" else np.asfortranarray(x_in)
+            y_in = y_in if order == "C" else np.asfortranarray(y_in)
+
+            x_gpu = gpuarray.to_gpu(x_in)
+            y_gpu = gpuarray.to_gpu(y_in)
+
+            np.testing.assert_allclose(gpuarray.stack((x_gpu, y_gpu), axis=axis).get(),
+                    np.stack((x_in, y_in), axis=axis))
+
+            np.testing.assert_allclose(gpuarray.stack((x_gpu, y_gpu), axis=axis).shape,
```

Comment:
- Don't recompute the stack results
- For the shape checking just use `assert` no need to bring in `assert_allclose`.